core/commands/pubsub.go

```diff
@@ -74,44 +75,17 @@ This command outputs data in the following encodings:
 		cmdkit.BoolOption("discover", "try to discover other peers subscribed to the same topic"),
 	},
 	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) {
-		n, err := cmdenv.GetNode(env)
+		api, err := cmdenv.GetApi(env)
 		if err != nil {
 			res.SetError(err, cmdkit.ErrNormal)
 			return
 		}
 
-		// Must be online!
-		if !n.OnlineMode() {
-			res.SetError(ErrNotOnline, cmdkit.ErrClient)
-			return
-		}
-
-		if n.Floodsub == nil {
-			res.SetError(fmt.Errorf("experimental pubsub feature not enabled. Run daemon with --enable-pubsub-experiment to use"), cmdkit.ErrNormal)
-			return
-		}
-
 		topic := req.Arguments[0]
-		sub, err := n.Floodsub.Subscribe(topic)
-		if err != nil {
-			res.SetError(err, cmdkit.ErrNormal)
-			return
-		}
-		defer sub.Cancel()
-
 		discover, _ := req.Options["discover"].(bool)
-		if discover {
-			go func() {
-				blk := blocks.NewBlock([]byte("floodsub:" + topic))
-				err := n.Blocks.AddBlock(blk)
-				if err != nil {
-					log.Error("pubsub discovery: ", err)
-					return
-				}
-
-				connectToPubSubPeers(req.Context, n, blk.Cid())
-			}()
-		}
+
+		sub, err := api.PubSub().Subscribe(req.Context, topic, options.PubSub.Discover(discover))
```

Comment:
This error seems unused/unchecked, sub's Next() call is checked soon after, but this should probably be handled here as well if we already have the err.